### PR TITLE
Add SDXL support to NeMo

### PR DESCRIPTION
**Pull request by Victor49152**
# What does this PR do ?

Add sdxl support 

**Collection**: NeMo multimodal

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [x] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Comment by Victor49152** [comment]
jenkins

**Comment by Victor49152** [comment]
jenkins

**Comment by ericharper** [comment]
jenkins

**Comment by Victor49152** [comment]
jenkins

**Comment by Victor49152** [comment]
jenkins

File: nemo/collections/nlp/models/language_modeling/megatron_base_model.py [hunk 1]
```diff
@@ -1215,12 +1216,27 @@ def _get_max_steps(self):
...
+                    frozen_submodules.append(module)
+            return frozen_submodule_names, frozen_submodules
+
         if self.use_fsdp:
             """ Top-evel FSDP model sharding """
             # Shard the top-level model hierarchically. We shard the strategy-unwrapped model not
             # to lose the structure of non-FSDP wrapped parameters (e.g, embedding)
             # TODO: Currently the main parameter data type is kept in fp32 (when O2=False). This needs to be
             # extended to support lower precision main parameters.
+            frozen_submodule_names, frozen_submodules = find_frozen_submodules(self.model)
```
[the remark this answers is not in the document]
Comment (Victor49152) [reply nemo/collections/nlp/models/language_modeling/megatron_base_model.py:1238]: Added comments

**Comment by Victor49152** [comment]
jenkins

**Review by yaoyu-33 — APPROVED** [review]

**Pull request merged**